"validates_start_with_lowercase" only applied to variable's name, but not to function's name

### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

As the title states, the rule option "validates_start_with_lowercase" only applied to the variable's name, but not to the function's name.

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
0.31.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
Homebrew
* Paste your configuration file:

```yml
identifier_name:
  allowed_symbols: 
    - _
  validates_start_with_lowercase: false
  min_length: 3
  max_length:
    warning: 50
    error: 60
  excluded: # excluded via string array
    - id
    - URL
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  No
* Which Xcode version are you using (check `xcode-select -p`)?
  Xcode 10.2
* Do you have a sample that shows the issue? 

```swift
public class Test {
    internal let _version = "1.2.3"
    internal func _start() { }
}
```

output:

> Loading configuration from '.swiftlint.yml'
> Linting Swift files at paths testlint.swift
> Linting 'testlint.swift' (1/1)
> /Users/troyl/SDKLite/airwatchsdk/testlint.swift:3:14: error: Identifier Name Violation: Function name should start with a lowercase character: '_start()' (identifier_name)
> Done linting! Found 1 violation, 1 serious in 1 file.

According to the output, the rule `validates_start_with_lowercase` applies to the variable name `_version`, but not to the function name `_start`. 

I do understand that mark the `_start` function as `private` will resolve this violation. But I have to argue that the behavior of `validates_start_with_lowercase` should be consistent and apply to all.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -49,7 +49,9 @@
 
 #### Bug Fixes
 
-* None.
+* Respect `validates_start_with_lowercase` value when linting function names
+  [Chris Brakebill](https://github.com/braker1nine)
+  [#2708](https://github.com/realm/SwiftLint/issues/2708)
 
 ## 0.49.1: Buanderie Principale
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -49,7 +49,7 @@
 
 #### Bug Fixes
 
-* Respect `validates_start_with_lowercase` value when linting function names
+* Respect `validates_start_with_lowercase` option when linting function names.  
   [Chris Brakebill](https://github.com/braker1nine)
   [#2708](https://github.com/realm/SwiftLint/issues/2708)
 

--- a/Source/SwiftLintFramework/Rules/Style/IdentifierNameRule.swift
+++ b/Source/SwiftLintFramework/Rules/Style/IdentifierNameRule.swift
@@ -72,7 +72,7 @@ public struct IdentifierNameRule: ASTRule, ConfigurationProviderRule {
             guard !firstCharacterIsAllowed else {
                 return []
             }
-            let requiresCaseCheck = configuration.validatesStartWithLowercase || isFunction
+            let requiresCaseCheck = configuration.validatesStartWithLowercase
             if requiresCaseCheck &&
                 kind != .varStatic && name.isViolatingCase && !name.isOperator {
                 let reason = "\(type) name should start with a lowercase character: '\(name)'"

--- a/Tests/SwiftLintFrameworkTests/IdentifierNameRuleTests.swift
+++ b/Tests/SwiftLintFrameworkTests/IdentifierNameRuleTests.swift
@@ -30,7 +30,8 @@ class IdentifierNameRuleTests: XCTestCase {
         let baseDescription = IdentifierNameRule.description
         let triggeringExamplesToRemove = [
             Example("↓let MyLet = 0"),
-            Example("enum Foo { case ↓MyEnum }")
+            Example("enum Foo { case ↓MyEnum }"),
+            Example("↓func IsOperator(name: String) -> Bool")
         ]
         let nonTriggeringExamples = baseDescription.nonTriggeringExamples +
             triggeringExamplesToRemove.removingViolationMarkers()

